src/main/java/kr/hellogsm/back_v2/domain/application/entity/grade/GraduateAdmissionGrade.java

```diff
@@ -64,6 +72,112 @@ public class GraduateAdmissionGrade extends AdmissionGrade {
 
     public GraduateAdmissionGrade(MiddleSchoolGrade middleSchoolGrade) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        GeneralScoreData result = null;
+        try {
+            result = objectMapper.readValue(middleSchoolGrade.getMiddleSchoolGradeText(), GeneralScoreData.class);
+        } catch (JsonProcessingException e) {
+            throw new ExpectedException("MiddleSchoolGrade값이 올바르지 않습니다", HttpStatus.BAD_REQUEST);
+        }
+
+        grade1Semester1Score = calc(result.score1_1(), result.system().equals("자유학년제") ? 54 : 18);
+        if (result.system().equals("자유학년제"))
+            grade1Semester2Score = calc(result.score1_2(), 54);
+        else
+            grade1Semester2Score = calc(result.score1_2(), result.freeSemester().equals("2-1") ? 36 : 18);
+        grade2Semester1Score = calc(result.score2_1(), result.system().equals("자유학년제") ? 54 : 36);
+        grade2Semester2Score = calc(result.score2_2(), result.system().equals("0") ? 36 : 54);
+        grade3Semester1Score = calc(result.score3_1(), 72);
+
+        BigDecimal generalCurriculumScoreSubtotal = Stream.of(
+                        grade1Semester1Score,
+                        grade1Semester2Score,
+                        grade2Semester1Score,
+                        grade2Semester2Score,
+                        grade3Semester1Score).reduce(BigDecimal.ZERO, BigDecimal::add)
+                .setScale(3, RoundingMode.HALF_UP);
+
+        // 예체능 성적
+        artisticScore = artSportsCalc(result.artSportsScore());
+
+        // 예체능 성적 + 교과 성적
+        curricularSubtotalScore = artisticScore
+                .add(generalCurriculumScoreSubtotal)
+                .setScale(4, RoundingMode.HALF_UP);
+
+        // 출결 성적
+        attendanceScore = calcAttendance(result.absentScore(), result.attendanceScore());
+
+        // 봉사 성적
+        volunteerScore = calcVolunteer(result.volunteerScore());
+
+        // 비 교과 성적
+        extracurricularSubtotalScore = attendanceScore.add(volunteerScore).setScale(4, RoundingMode.HALF_UP);
+
+        totalScore = curricularSubtotalScore.add(extracurricularSubtotalScore).setScale(3, RoundingMode.HALF_UP);
+
+        percentileRank = calcPercentileRank(totalScore);
+    }
+
+    private BigDecimal calc(List<BigDecimal> scoreArray, int maxPoint) {
+        if (scoreArray == null) return BigDecimal.valueOf(0);
+        // (A 개수 * 5) + (B 개수 * 4) + (C 개수 * 3) + (D 개수 * 2) + (E 개수 * 1)
+        BigDecimal reduceResult = scoreArray.stream().reduce(BigDecimal.ZERO, BigDecimal::add);
+        // reduceResult / (과목 수 * 5) (소수점 6째자리 반올림)
+        BigDecimal divideResult = reduceResult.divide(BigDecimal.valueOf(scoreArray.size() * 5L), 5, RoundingMode.HALF_UP);
+        // 각 학년 당 배점 * divideResult (소수점 4째자리에서 반올림)
+        return divideResult.multiply(BigDecimal.valueOf(maxPoint)).setScale(3, RoundingMode.HALF_UP);
+    }
+
+    private BigDecimal artSportsCalc(List<BigDecimal> scoreArray) {
```

Comment:
다른 계산 관련 메서드는 `calc~` 처럼 calc이 맨 앞에 있는데
`artSportsCalc()` 하나만 `calc` 이 맨 뒤에 있네요.

일부로 의도한게 아니라면 아래 suggestion 처럼 바꾸는게 일관성 있어보여요.

```suggestion
    private BigDecimal calcArtSports(List<BigDecimal> scoreArray) {
```

Reply:
실수 인것 같네요
수정 완료했습니다~